### ISSUE TITLE
[CML][PM] White noise when being runtime suspended after playback

### ISSUE THREAD

**Issue by xun2z** [thesofproject/linux#603]
**Summary:**  
when playback using alc5682 codec on cometlake, there is a about 2s white noise when entering runtime suspend.
  
**Test steps:**
1.  Play some music, eg.
`aplay -vv -Dhw:0,0 -c2 -d2 48California_Dreaming.wav`
2. when it ends, check the runtime status
`cat /sys/bus/pci/devices/0000:00:1f.3/power/runtime_status`
The noise occurs when entering runtime suspended.
However,  this noise only happens in suspending and if we trigger new playback, there will be no noise during playback.

**Test env:**
sof cml-001-drop-stable: 1f5a9657114
kernel topic/cros-sof-v4.19: f69d618
tplg: sof-cnl-rt5682.tplg

noise file: [noise.zip](https://github.com/thesofproject/linux/files/2806681/noise.zip)

**Comment by bardliao** [comment]
MCLK is turned off in runtime suspended. However, sof enter runtime suspend before DAPM off. In other word, MCLK is turned off before codec is powered off. And noise will occur in that period.

**Comment by plbossart** [comment]
> MCLK is turned off in runtime suspended. However, sof enter runtime 
> suspend before DAPM off. In other word, MCLK is turned off before 
> codec is powered off. And noise will occur in that period.
>
I believe we will have to make the MCLK/BCLK controllable from the 
machine/codec driver. This is a feature that was implemented in the cAVS 
driver and that we know we need to avoid issues due to clock 
transitions, but haven't done any work on. There are some fields in the 
IPC that could be used to leave the MCLK on, but they are not set. see 
SOF issue #629 https://github.com/thesofproject/sof/issues/629

We'd also need the pm_runtime to suspend the DSP when there is no need 
for any clocks.

In other words it's not going to be fixed overnight.

**Comment by emilchudzik** [comment]
@wwittbrx bringup Hatch. thank you Wojtek.

We just checked flow from topic and we haven't repro. There is no "white noises" after end of the playback on each time.

Configuration:
Platform: Hatch CML
kernel: cros-sof-v4.19 / [8492020d5967](https://github.com/thesofproject/linux/commit/8492020d5967) + PR#651 applied
FW: cml-002-drop-stable / [dcd825b](https://github.com/thesofproject/sof/commit/dcd825b923a680f63adde1b392638a14463ead63)
tplg: sof-cnl-rt5682.tplg

Checked also on WHL RVP and there is also no reproduction with "while noises":
kernel: cros-sof-v4.19 / [8492020d5967](https://github.com/thesofproject/linux/commit/8492020d5967) + PR#651 applied
FW: cml-002-drop-stable / [dcd825b](https://github.com/thesofproject/sof/commit/dcd825b923a680f63adde1b392638a14463ead63)
tplg for hda: sof-hda-generic.tplg

**Comment by bardliao** [comment]
@emilchudzik Did you test it by headphone?

**Comment by emilchudzik** [comment]
No, I tested it by hdmi.
On headphone I haven't any sound although there is no errors in dmesg log and aplay show playback progress without any issues. I haven't went deeper for debugging it yet - I will try it again with deeper analysis and I will give a feedback.

**Comment by emilchudzik** [comment]
I just tested flow from topic on headphone (thanks to @bardliao to help configure alsamixer).

On SSP0 (connected headphones) I have this observation:
- I play 5 seconds: `aplay -vv -Dhw:0,0 sine.wav -d5`
- after that I hear about 1 seconds of silence
- then I hear about 1 seconds of something like mic is opened, but this is not "uggly white noise", but just something like "simple noise from mic"

Here is a dmesg from start aplay to going suspended
[dmesg.txt](https://github.com/thesofproject/linux/files/2914271/dmesg.txt)

**Comment by plbossart** [comment]
@xun2z is this issue still relevant? It's been 24 days without update.

**Comment by xun2z** [comment]
@plbossart We still have this issue. Bard says he's trying to set a `ignore_pmdown_time` flag, which can shut down rt5682 immediately after playback. I'll look into it and help to check with Bard.

**Comment by xun2z** [comment]
close since #741 is the fix
